### ISSUE TITLE
Implement external authentication with Northstar

### ISSUE THREAD

**Issue by aaronschachter** [DoSomethingArchive/legacy-website#4526]
In looking to build the ["Change Email" screen in the iOS app](https://github.com/DoSomething/LetsDoThis-iOS/issues/21), I've run into a pickle.  The PUT to the Northstar User works fine and changes the email.  The Drupal User is not updated.

The opposite situation is true as well, if I update my email within Drupal, my Northstar email is not updated.

We'll run into this issue with Mobile Number and Password as well.

Not sure what best course of action is, but I believe this is functionality that @sergii-tkachenko has gotten working for the DS Canada and DS UK sites, by building the [External Auth module](https://github.com/DoSomething/drupal-external-auth), which we've talked about releasing as a contrib module on drupal.org. The module queries a third party service in order to authenticate, and sync the Drupal User with the authenticated user returned in the response.

One approach is that we implement the module for connecting to Northstar, to avoid building out custom Services calls (Northstar doing a PUT on a Drupal User, and vice versa) that would seem to duplicate the work @sergii-tkachenko has already gotten in place.

DS Canada implementation: https://github.com/DoSomething/dosomething/tree/dev/lib/modules/dosomething/dosomething_canada/includes

DS UK implementation: 
https://github.com/DoSomething/dosomething/tree/dev/lib/modules/dosomething/dosomething_uk/includes

cc @jonuy @angaither @DFurnes

**Comment by jonuy** [comment]
A solution Andrea was working on before was to forward Drupal logins, registrations, and updates onto Northstar. There's a Northstar module (https://github.com/DoSomething/dosomething/pull/4434) that seems to just be waiting on Guzzle to be available. What are people's thoughts on this? Can this be a viable nearer-term solution?

Forwarding user updates from Northstar onto Drupal seems trivial and is something we're doing currently for user registrations.

Anyone know what amount of work is required to use the external auth module as a solution? What kind of dev is needed? I assume we'd also need to migrate all users from Drupal over to Northstar too before the flip?

**Comment by aaronschachter** [comment]
@sergii-tkachenko -- you know the External Auth module best, can you weigh in?  

@jonuy Forwarding updates from Northstar to Drupal would still require dev work either on the Northstar end or Drupal end to transform with the various User field arrays.

**Comment by aaronschachter** [comment]
It sounds like we may be able to get by in Version 1 of the mobile app without syncing shared fields such as Email, Password, and Mobile.. so, it may be worth investing in the best solution vs the quickest here.

**Comment by angaither** [comment]
The northstar module still exists in drupal, and it's running.
Currently as people login it sends a request to forward their user info into NS. It's just using `drupal_http_request` currently

**Comment by sergiitk** [comment]
Can we setup a meeting to discuss all this matters?

**Comment by aaronschachter** [comment]
Follow up point is that additionally, besides the User's account/profile info, her Signups and Reportbacks also need to be synced with Northstar.

**Comment by jonuy** [comment]
Follow up on a meeting with Aaron, Andrea, and Sergii: The external auth module does sound like the solution we'd want to use for getting Drupal to handle authentication with Northstar. And now that our servers are on a more updated version of php, Guzzle can come into play here too. This isn't a solution we need in place right now right now for current products. Most of our syncing issues should be covered with work already done with the northstar module and upcoming work in #4570 and #4571.

As for this, we should talk to Matt and product about what priority should be.

**Comment by angaither** [comment]
cc @DFurnes

**Comment by angaither** [comment]
@DFurnes :heavy_check_mark:ing in on this, how is the verify creds working? are we getting :white_check_mark: or :x:

refs #6231

**Comment by DFurnes** [comment]
Just took a peek. There are quite a few where validation failed and `NONE` was set in the Northstar ID profile field... but yeah, there's a good chance this is because of #6231 so not totally unexpected.

**Comment by DFurnes** [comment]
Oh man this is _done_ 💥
